src/Symfony/Component/Console/Application.php

```diff
@@ -1169,4 +1169,27 @@ private function splitStringByWidth($string, $width)
 
         return $lines;
     }
+
+    /**
+     * Returns all namespaces of the command name.
+     *
+     * @param string $name The full name of the command
+     *
+     * @return array The namespaces of the command
+     */
+    private function extractAllNamespaces($name)
+    {
+        $parts = explode(':', $name, -1);
```

Comment:
`extractNamespace` is okay, but comments sometimes don't sync properly with the evolution of the code. My suggestion would be to apply extract method to the line containing the `explode()`, wich seems to need additional semantics, and name the method in consonance with the comment you just wrote. 

For example:

```
$parts = $this->explodeSkippingCommandName($name);
```